### PR TITLE
[Docs] Hedging context

### LINKED ISSUE #1727: [Clarification]: Hedging contexts
### Is your feature request related to a specific problem? Or an existing feature?

While I was playing with hedging strategy I realized that we have `PrimaryContext` and `ActionContext` on [`OnHedgingArguments<TResult>` struct](https://github.com/App-vNext/Polly/blob/main/src/Polly.Core/Hedging/OnHedgingArguments.cs).

It is unclear:
- what is difference between the two 
  - I did not find the documentation comment really helpful
- when to use which
  - hedging documentation does not even mention primary only the `ActionContext` 

### Describe the solution you'd like

I think we could improve on the documentation comments also on the hedging documentation page.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by martintmk**
# Pull Request

## The issue or feature being addressed

Closes #1727 

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/strategies/hedging.md [hunk 1]
```diff
@@ -440,6 +440,40 @@ sequenceDiagram
     P->>C: Propagates failure (R3)
 ```
 
+## Hedging and resilience context
+
+The hedging strategy supports the concurrent execution and cancellation of multiple hedged actions. To safeguard against concurrency issues when these actions access the same `ResilienceContext`, the hedging strategy ensures that each hedged action has its unique `ResilienceContext`. The hedging strategy distinguishes between two types of contexts:
```
Comment (martincostello) [inline docs/strategies/hedging.md:445]:
```suggestion
The hedging strategy supports the concurrent execution and cancellation of multiple hedged actions. To safeguard against concurrency issues when these actions access the same `ResilienceContext`, the hedging strategy ensures that each hedged action has its own unique `ResilienceContext`. The hedging strategy distinguishes between two types of contexts:
```

File: docs/strategies/hedging.md [hunk 2]
```diff
@@ -440,6 +440,40 @@ sequenceDiagram
...
+
+- **Primary context**: The original resilience context that the hedging strategy receives.
+- **Action context**: A deep copy of the primary context with distinct cancellation token for a single hedged action.
```
Comment (martincostello) [inline docs/strategies/hedging.md:448]:
```suggestion
- **Action context**: A deep copy of the primary context with a distinct cancellation token for a single hedged action.
```

File: docs/strategies/hedging.md [hunk 3]
```diff
@@ -440,6 +440,40 @@ sequenceDiagram
...
+
+Here's the flow:
+
+- The strategy gets the primary context and creates a snapshot for deep cloning.
+- For each hedged action execution, the hedging strategy makes a deep copy of the original context. The deep copy has its own cancellation token designated for that execution. Note that the first execution (primary) uses the original resilience context, albeit with a cloned set of resilience properties.
+- After the strategy has an accepted result from a hedged action, the resilience context from the action is merged back into the primary context.
+- All ongoing hedged actions are cancelled and discarded. The hedging strategy awaits the propagation of cancellation.
+
+### Merging action context into primary context
```
Comment (martincostello) [inline docs/strategies/hedging.md:457]:
```suggestion
### Merging action context into the primary context
```

File: docs/strategies/hedging.md [hunk 4]
```diff
@@ -440,6 +440,40 @@ sequenceDiagram
...
+
+After merging, the primary context contains:
+
+- New properties saved during the action context's execution.
```
Comment (martincostello) [inline docs/strategies/hedging.md:461]:
```suggestion
- New properties created during the action context's execution.
```

**Review by martincostello — APPROVED** [review]

File: docs/strategies/hedging.md [hunk 5]
```diff
@@ -440,6 +440,40 @@ sequenceDiagram
...
+- Existing properties updated during execution.
+
+> [!NOTE]
+> If you delete a property from the action context and this property exists in the primary context, the change is ignored, and the primary context retains the original property.
```
Comment (peter-csala) [inline docs/strategies/hedging.md:465]: As far as I can see you can't delete a property from the context. Only `GetValue`, `TryGetValue` and `Set` are exposed publicly. I've used the term soft-delete intentionally inside the issue to describe a case when you set a proprety value to `null`.

File: docs/strategies/hedging.md [hunk 6]
```diff
@@ -440,6 +440,40 @@ sequenceDiagram
...
 ```
 
+## Hedging and resilience context
+
+The hedging strategy supports the concurrent execution and cancellation of multiple hedged actions. To safeguard against concurrency issues when these actions access the same `ResilienceContext`, the hedging strategy ensures that each hedged action has its unique `ResilienceContext`. The hedging strategy distinguishes between two types of contexts:
+
+- **Primary context**: The original resilience context that the hedging strategy receives.
+- **Action context**: A deep copy of the primary context with distinct cancellation token for a single hedged action.
+
+Here's the flow:
```
Comment (peter-csala) [inline docs/strategies/hedging.md:450]: I was thinking whether it make sense to add a mermaid sequence diagram here as well. + also including `OnHedging` and `ActionGenerator`. 

What do you think?

File: docs/strategies/hedging.md (hunk 5 above)
Comment (martintmk) [reply docs/strategies/hedging.md]: You are right, dropped this section and updated the list with this info.

File: docs/strategies/hedging.md (hunk 6 above)
Comment (martintmk) [reply docs/strategies/hedging.md:450]: Sure, it could be useful, let's do it in follow-up. 

Can you give us a hand here?

Comment (peter-csala) [reply docs/strategies/hedging.md:450]: Yepp, sure 👍

**Pull request merged**